### PR TITLE
fix(ci): run and build right e2e tests artifacts on e2e-test-main

### LINKED ISSUE #8808: e2e-test-main workflow requires to add github token in order to build podman desktop for win-update-e2e-test correctly
### Bug description

Ie.: https://github.com/containers/podman-desktop/actions/runs/10773962330/job/29874913862

Error:
```
Skipping code signing, no environment variables set for that.
Skipping code signing, no environment variables set for that.
  • building block map  blockMapFile=dist\podman-desktop-1.0.0-setup-x64.exe.blockmap
  ⨯ Cannot cleanup: 

Error #1 --------------------------------------------------------------------------------
Error: GitHub Personal Access Token is not set, neither programmatically, nor using env "GH_TOKEN"
    at new GitHubPublisher (D:\a\podman-desktop\podman-desktop\node_modules\app-builder-lib\node_modules\electron-publish\src\gitHubPublisher.ts:52:15)
    at createPublisher (D:\a\podman-desktop\podman-desktop\node_modules\app-builder-lib\src\publish\PublishManager.ts:303:14)
    at PublishManager.getOrCreatePublisher (D:\a\podman-desktop\podman-desktop\node_modules\app-builder-lib\src\publish\PublishManager.ts:222:19)
    at PublishManager.scheduleUpload (D:\a\podman-desktop\podman-desktop\node_modules\app-builder-lib\src\publish\PublishManager.ts:152:28)
    at PublishManager.artifactCreatedWithoutExplicitPublishConfig (D:\a\podman-desktop\podman-desktop\node_modules\app-builder-lib\src\publish\PublishManager.ts:202:14)
```

### Operating system

Windows

### Installation Method

Other

### Version

next (development version)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Add GITHUB_TOKEN into windows workflow to allow building signed artifacts and restrcts e2e test targets to be run on linux.
### What issues does this PR fix or reference?
#8808

### How to test this PR?
`pnpm test:e2e` will run all tests in `src/specs/*.spec.ts` but not tests in `src/specs/installation/`
`pnpm test:e2e` will run tests from `src/specs/installation/*` only

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — APPROVED** [review]

File: tests/playwright/src/specs/installation/update-install.spec.ts [hunk 1]
```diff
@@ -42,6 +44,7 @@ test.afterAll(async ({ runner }) => {
 });
 
 test.describe.serial('Podman Desktop Update Update installation offering', () => {
+  test.skip(isLinux, 'Update is not supported on Linux');
```
Comment (cbr7) [inline tests/playwright/src/specs/installation/update-install.spec.ts]: Move test.skip out of the describe, this way the check is validated before the beforeAll hook is run,

File: package.json [hunk 2]
```diff
@@ -38,7 +38,7 @@
     "test:unit": "npm run test:main && npm run test:preload && npm run test:preload-docker-extension && npm run test:preload-webview && npm run test:ui && npm run test:renderer && npm run test:scripts:stylesheet && npm run test:tools && npm run test:extensions",
     "test:e2e": "npm run test:e2e:build && npm run test:e2e:run",
     "test:e2e:build": "cross-env NODE_ENV=development MODE=development DEBUG=pw:browser npm run build",
-    "test:e2e:run": "xvfb-maybe --auto-servernum --server-args='-screen 0 1280x960x24' -- npx playwright test tests/playwright/src/specs/",
+    "test:e2e:run": "xvfb-maybe --auto-servernum --server-args='-screen 0 1280x960x24' -- npx playwright test tests/playwright/src/specs/*.spec.ts",
```
Comment (cbr7) [inline package.json]: Won't this have consequences for windows tests? From what I remember the wildcard does not work on windows. Wouldn't it be easier to move the installation spec in another folder, like `src/installation/specs/` ?

Alternatively you can use the test tagging function of the playwright framework, like it is for the smoke tests.

EDIT: Regarding the tagging, you can tag the installation test suite with some tag as you want and then change the `test:e2e:run` command to execute an inverted grep test running `npx playwright test tests/playwright/src/specs/ --grep-invert @<tag of installation test suite>` to run all tests except those that match the tag and then in the command that you want to actually run the test you make it `npx playwright test tests/playwright/src/specs/ -g @<tag of installation test suite>` as is for the smoke test.

**Review by cbr7 — CHANGES_REQUESTED** [review]
Left some comments.

File: package.json (hunk 2 above)
Comment (odockal) [reply package.json]: @cbr7 yes, it still does not work. So the tag it is then.

**Review by cbr7 — APPROVED** [review]

**Pull request merged**